beacon-chain/sync/custody.go

```diff
@@ -103,12 +103,35 @@ func (s *Service) custodyGroupCount(context.Context) (uint64, error) {
 		return cfg.NumberOfCustodyGroups, nil
 	}
 
+	// Calculate validator custody requirements
 	validatorsCustodyRequirement, err := s.validatorsCustodyRequirement()
 	if err != nil {
 		return 0, errors.Wrap(err, "validators custody requirement")
 	}
 
-	return max(cfg.CustodyRequirement, validatorsCustodyRequirement), nil
+	effectiveCustodyRequirement := max(cfg.CustodyRequirement, validatorsCustodyRequirement)
+
+	// If we're not in semi-supernode mode, just use the effective requirement.
+	if !flags.Get().SemiSupernode {
+		return effectiveCustodyRequirement, nil
+	}
+
+	// Semi-supernode mode custodies exactly half of all custody groups (e.g. 64 out of 128),
+	// which is the minimum required for blob reconstruction.
+	semiSupernodeTarget := cfg.NumberOfCustodyGroups / 2
+	if effectiveCustodyRequirement <= semiSupernodeTarget {
+		return semiSupernodeTarget, nil
+	}
+
+	// If effective requirement is higher than semi-supernode target, use it.
+	log.WithFields(logrus.Fields{
+		"semiSupernodeTarget":          semiSupernodeTarget,
+		"effectiveCustodyRequirement":  effectiveCustodyRequirement,
+		"custodyRequirement":           cfg.CustodyRequirement,
+		"validatorsCustodyRequirement": validatorsCustodyRequirement,
+	}).Warn("Custody requirements exceed semi-supernode mode; using higher custody count to meet requirements.")
```

Comment:
Is this worth a WARN level log? If running with many validators, this log message would be firing often?

Reply:
This is a good point, I think it'll fire off 1once a minute after reivew I should make sure it only triggers once

Reply:
after trying a few things it seems overly complex for this, just turned it to debug. let me know if it's preferred to just remove it.

Reply:
i ended up removing it https://github.com/OffchainLabs/prysm/pull/16029/commits/a6a67b5427135a8392fe6619c6b4539fe529b705